gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/diagnosticreport/SpecimenMapperTest.java

```diff
@@ -162,6 +242,35 @@ private static DiagnosticReport generateDiagnosticReportWithNoSpecimenReference(
         return (DiagnosticReport) new DiagnosticReport().setId(DR_ID);
     }
 
+    private void assertMetaSecurityIsPresent(final Meta meta) {
+        final List<Coding> metaSecurity = meta.getSecurity();
+        final int metaSecuritySize = metaSecurity.size();
+        final Coding metaSecurityCoding = metaSecurity.get(0);
+        final UriType metaProfile = meta.getProfile().get(0);
+
+        assertAll(
+            () -> assertThat(metaSecuritySize).isEqualTo(1),
+            () -> assertThat(metaProfile.getValue()).isEqualTo(SPECIMEN_META_PROFILE),
+            () -> assertThat(metaSecurityCoding.getCode()).isEqualTo(NOPAT_CV.getCode()),
+            () -> assertThat(metaSecurityCoding.getDisplay()).isEqualTo(NOPAT_CV.getDisplayName()),
+            () -> assertThat(metaSecurityCoding.getSystem()).isEqualTo(NOPAT_CV.getCodeSystem()));
+    }
+
+    private RCMRMT030101UKCompoundStatement getCompoundStatement(RCMRMT030101UKEhrExtract ehrExtract) {
+        return ehrExtract.getComponent().get(0)
+            .getEhrFolder().getComponent().get(0)
+            .getEhrComposition().getComponent().get(0)
+            .getCompoundStatement().getComponent().get(0)
+            .getCompoundStatement();
+    }
+
+    private RCMRMT030101UKEhrComposition getEhrComposition(RCMRMT030101UKEhrExtract ehrExtract) {
+
+        return ehrExtract.getComponent().get(0)
+            .getEhrFolder().getComponent().get(0)
+            .getEhrComposition();
+    }
```

Comment:
Again, would we be able to abstract this out into `TestUtility` perhaps?

Furthermore, it seems that at the `EHR Extract` level there can me `1 .. *` components, each component containing one and only one `EHR Folder`. Within the `EHR Folder` there, again, can be `1 .. *` components, each of these components containing one and only one `EHR Composition`.

Therefore we may be able to make this more reusable 

```suggestion
    private RCMRMT030101UKEhrComposition getEhrComposition(RCMRMT030101UKEhrExtract ehrExtract, int ehrExtractComponentIndex) {

        return ehrExtract.getComponent().get(ehrExtractComponentIndex)
            .getEhrFolder().getComponent().getFirst()
            .getEhrComposition();
    }
```

Reply:
Yes, good suggestion and worth considering 👍 . As mentioned before, I am planning to bring all those refactoring and optimizations in a following clean-up PR otherwise with quite a few business logic related changes it might be quite confusing and difficult to review .